llama_index/chat_engine/types.py

```diff
@@ -20,13 +20,14 @@ class AgentChatResponse:
 
     response: str = ""
     sources: List[ToolOutput] = field(default_factory=list)
-    _nodes: List[NodeWithScore] = None
+    _nodes: List[NodeWithScore] = field(default_factory=list)
+
     @property
     def source_nodes(self) -> List[NodeWithScore]:
         if self._nodes is None:
```

Comment:
This should probably be `if not self._nodes:` because of line 23 using a default factory of list.